data/transactions/logic/assembler.go

```diff
@@ -1428,16 +1438,16 @@ func typeStores(pgm *ProgramKnowledge, args []string) (StackTypes, StackTypes, e
 }
 
 func typeLoad(pgm *ProgramKnowledge, args []string) (StackTypes, StackTypes, error) {
-	scratchIndex, ok := getImm(args, 0)
+	scratchIndex, ok := getImm(args, 0, false)
 	if !ok {
 		return nil, nil, nil
 	}
 	return nil, StackTypes{pgm.scratchSpace[scratchIndex]}, nil
```

Comment:
so it seems we previously allowed people to pass in `-1` from `getImm`, and passed in `scratchSpace`, thus panic?

Reply:
Yeah, I often forget how fragile the `type*` methods are. A lot of checking happens later, so I forget that these need good bounds checking on their own.  Someday, I'd like to refactor so they are a bit safer/easier to write.